test/hpr_packet_reporter_SUITE.erl

```diff
@@ -0,0 +1,187 @@
+-module(hpr_packet_reporter_SUITE).
+
+-export([
+    all/0,
+    init_per_testcase/2,
+    end_per_testcase/2
+]).
+
+-export([
+    upload_report_test/1,
+    upload_window_test/1
+]).
+
+-include("hpr.hrl").
+-include("../src/grpc/autogen/server/packet_router_pb.hrl").
+-include("../src/grpc/autogen/server/config_pb.hrl").
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-record(state, {
+    aws_client :: aws_client:aws_client(),
+    write_dir :: string(),
+    file_path :: string(),
+    max_file_size :: non_neg_integer(),
+    report_interval :: timer:tref() | undefined,
+    interval_duration :: non_neg_integer() | undefined,
+    upload_window :: non_neg_integer(),
+    upload_window_start_time :: non_neg_integer(),
+    bucket :: binary()
+}).
+
+%%--------------------------------------------------------------------
+%% COMMON TEST CALLBACK FUNCTIONS
+%%--------------------------------------------------------------------
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%   Running tests for this suite
+%% @end
+%%--------------------------------------------------------------------
+all() ->
+    [
+        upload_report_test,
+        upload_window_test
+    ].
+
+%%--------------------------------------------------------------------
+%% TEST CASE SETUP
+%%--------------------------------------------------------------------
+init_per_testcase(TestCase, Config) ->
+    Config1 = test_utils:init_per_testcase(TestCase, Config),
+    meck:new(aws_s3, [passthrough]),
+    meck:new(hpr_packet_reporter, [passthrough]),
+
+    Config1.
+
+%%--------------------------------------------------------------------
+%% TEST CASE TEARDOWN
+%%--------------------------------------------------------------------
+end_per_testcase(TestCase, Config) ->
+    ets:delete_all_objects(hpr_packet_report_ets),
+    meck:unload(aws_s3),
```

Comment:
`meck:unload/0` unloads all mocked modules. This might be safer in case a test ends up mocking a module.